app/storage/s3ResourceHandler.js

```diff
@@ -19,91 +19,113 @@
 const conf = require('./../conf.js').conf;
 const S3ClientClass = require('./s3NewClient');
 const cipher = require('./cipher');
+const _ = require('lodash');
 
 class S3ResourceHandler {
 
-  constructor(logger, resourceKey, bucketName, location, endpoint) {
+  constructor(args) {
+    let { logger, path, bucketName=null, bucketConfObj=null, location, endpoint, org } = args;
+    if(!bucketConfObj && !bucketName){
+      throw new Error('Pass a bucketConfObj or bucketName');
+    }
+    bucketConfObj = bucketConfObj || {};
+    if(location){
+      bucketConfObj.location = location;
+    }
+    if(!bucketName){
+      bucketName = conf.storage.getResourceBucket(location);
+    }
```

Comment:
```suggestion
```

Reply:
We support multiple different storage handlers. That function's for those using `s3`. The new one that this PR creates is `s3OrgBucket`. I had to create the new one so the old one can continue to handle existing db data properly.